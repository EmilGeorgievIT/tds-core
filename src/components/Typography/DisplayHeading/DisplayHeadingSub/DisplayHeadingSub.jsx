import React from 'react'
import PropTypes from 'prop-types'
import safeRest from '../../../../safeRest'

import styles from '../DisplayHeading.modules.scss'

/**
 * Subscript text for headings.
 */
const DisplayHeadingSub = ({ children, ...rest }) => (
  <sub {...safeRest(rest)} className={styles.sub}>{children}</sub>
)

DisplayHeadingSub.propTypes = {
  children: PropTypes.string.isRequired
}

export default DisplayHeadingSub
