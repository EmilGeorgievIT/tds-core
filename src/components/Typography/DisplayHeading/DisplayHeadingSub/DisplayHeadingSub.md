```
<div>
  <DisplayHeading>Great Deals<DisplayHeading.Sub>1</DisplayHeading.Sub></DisplayHeading>
</div>
```
