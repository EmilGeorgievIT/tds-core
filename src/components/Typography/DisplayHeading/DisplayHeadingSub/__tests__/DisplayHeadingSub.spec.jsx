import React from 'react'
import { shallow } from 'enzyme'


import DisplayHeadingSub from '../DisplayHeadingSub'

describe('DisplayHeadingSub', () => {
  const doShallow = props => (
    shallow(<DisplayHeadingSub {...props}>Some content</DisplayHeadingSub>)
  )

  it('renders', () => {
    const displayHeadingSub = doShallow()

    expect(displayHeadingSub).toMatchSnapshot()
  })

  it('renders an HTML sub tag', () => {
    const displayHeadingSub = doShallow()

    expect(displayHeadingSub).toHaveTagName('sub')
  })

  it('passes additional attributes to the sup element', () => {
    const displayHeadingSub = doShallow({ id: 'the-sub-text' })

    expect(displayHeadingSub).toHaveProp('id', 'the-sub-text')
  })

  it('does not allow custom CSS', () => {
    const displayHeadingSub = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(displayHeadingSub).not.toHaveProp('className', 'my-custom-class')
    expect(displayHeadingSub).not.toHaveProp('style')
  })
})
