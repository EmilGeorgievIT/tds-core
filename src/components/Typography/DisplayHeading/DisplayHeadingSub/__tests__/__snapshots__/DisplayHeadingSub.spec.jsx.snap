// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayHeadingSub renders 1`] = `
<sub
  className="sub"
>
  Some content
</sub>
`;
